python/lib/sift_py/calculated_channels/config.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, ConfigDict, field_validator, model_validator
+from typing_extensions import NotRequired, TypedDict
+
+from sift_py.rule.config import (
+    ExpressionChannelReference,
+    ExpressionChannelReferenceChannelConfig,
+    _channel_references_from_dicts,
+)
+
+
+class CalculatedChannelConfig(BaseModel):
+    """
+    Configuration for a calculated channel.
+
+    - `name`: Name of the calculated channel.
+    - `description`: Description of the calculated channel.
+    - `expression`: The expression to calculate the value of the calculated channel.
+    - `channel_references`: A list of channel references that are used in the expression. Each reference can either
+      be an `ExpressionChannelReference` or `ExpressionChannelReferenceChannelConfig`.
+    - `units`: Units of the calculated channel.
+    - `client_key`: A user defined string that uniquely identifies the calculated channel. Name is unique, but may be changed.
+    - `asset_ids`: A list of asset names to make the calculation available for.
+    - `tag_ids`: A list of tag_ids on assets to make the calculation available for.
+    - `all_assets`: A flag that, when set to `True`, associates the calculated channel with all assets.
+    """
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    name: str
+    description: str = ""
+    expression: str
+    channel_references: List[
+        Union[ExpressionChannelReference, ExpressionChannelReferenceChannelConfig]
+    ]
+    units: Optional[str] = None
+    calculated_channel_id: Optional[str] = None
+    client_key: Optional[str] = None
```

Comment:
We generally want to encourage users to use client keys moving forward at this point, I would see if we could consider making the client key field non-optional. I don't think there's any reason a user wouldn't use a client key at this point. If I'm not mistaken, client key was kept optional in most of our other apis for backwards compatibility reasons, which I don't think we need to consider here since this service is new.

Reply:
Chatted about this offline -- we're opening some tickets to improve the UI related to this.